moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -377,32 +378,31 @@ bool RuckigSmoothing::runRuckig(robot_trajectory::RobotTrajectory& trajectory,
   return true;
 }
 
-void RuckigSmoothing::extendTrajectoryDuration(const double duration_extension_factor, size_t num_waypoints,
+void RuckigSmoothing::extendTrajectoryDuration(const double duration_extension_factor, size_t waypoint_idx,
                                                const size_t num_dof, const std::vector<int>& move_group_idx,
                                                const robot_trajectory::RobotTrajectory& original_trajectory,
                                                robot_trajectory::RobotTrajectory& trajectory)
 {
-  for (size_t time_stretch_idx = 1; time_stretch_idx < num_waypoints; ++time_stretch_idx)
+  trajectory.setWayPointDurationFromPrevious(waypoint_idx + 1,
+                                             duration_extension_factor *
+                                                 original_trajectory.getWayPointDurationFromPrevious(waypoint_idx + 1));
+  // re-calculate waypoint velocity and acceleration
+  auto target_state = trajectory.getWayPointPtr(waypoint_idx + 1);
+  const auto prev_state = trajectory.getWayPointPtr(waypoint_idx);
+
+  double timestep = trajectory.getWayPointDurationFromPrevious(waypoint_idx + 1);
+
+  for (size_t joint = 0; joint < num_dof; ++joint)
   {
-    trajectory.setWayPointDurationFromPrevious(
-        time_stretch_idx,
-        duration_extension_factor * original_trajectory.getWayPointDurationFromPrevious(time_stretch_idx));
-    // re-calculate waypoint velocity and acceleration
-    auto target_state = trajectory.getWayPointPtr(time_stretch_idx);
-    const auto prev_state = trajectory.getWayPointPtr(time_stretch_idx - 1);
-    double timestep = trajectory.getAverageSegmentDuration();
-    for (size_t joint = 0; joint < num_dof; ++joint)
-    {
-      target_state->setVariableVelocity(move_group_idx.at(joint),
-                                        (1 / duration_extension_factor) *
-                                            target_state->getVariableVelocity(move_group_idx.at(joint)));
+    target_state->setVariableVelocity(move_group_idx.at(joint),
+                                      (1 / duration_extension_factor) *
+                                          target_state->getVariableVelocity(move_group_idx.at(joint)));
 
-      double prev_velocity = prev_state->getVariableVelocity(move_group_idx.at(joint));
-      double curr_velocity = target_state->getVariableVelocity(move_group_idx.at(joint));
-      target_state->setVariableAcceleration(move_group_idx.at(joint), (curr_velocity - prev_velocity) / timestep);
-    }
-    target_state->update();
+    double prev_velocity = prev_state->getVariableVelocity(move_group_idx.at(joint));
+    double curr_velocity = target_state->getVariableVelocity(move_group_idx.at(joint));
+    target_state->setVariableAcceleration(move_group_idx.at(joint), (curr_velocity - prev_velocity) / timestep);
   }
+  target_state->update();
```

Comment:
just curious, why do we care about updated transforms at all? This looks like a redundant step that only slows down computation here. Positions are also not changed, so there are no transforms to update.

Reply:
From what I understand from documentation.
Update() only needs to be called before there is any kind of collision checking.
Which doesn't happen here. 
And like you mentioned , no positions have changed requiring a change in transform.That would mean the update () call here can be removed.

@AndyZe  is this correct.?

Reply:
Built and tested without the update, doesn't seem to cause any issues within the tests.

Reply:
Done.